### PR TITLE
Add P-256 and P-384 in cryptographic signatures

### PR DESCRIPTION
**Pull request by troyronda**
Closes: #66

### CODE REVIEW

**Review by quartzjer — APPROVED** [review]

File: spec/spec.md [hunk 1]
```diff
@@ -597,6 +597,8 @@ While ION supports any public key JWK representation in a DID Document, implemen
 |--------|-------------|
 |secp256k1|ES256K|
 |Ed25519|EdDSA|
+|P-256 (secp256r1)|ES256|
+|P-384 (secp384r1)|ES384|
```
Comment (Sakurann) [inline spec/spec.md:601]: Suggest we make P curves optional for both verifiers and wallets.
and suggest we enhance the table as below:

|Key Type|JWT Algorithm|Mandatory/Optional for the Verifier|Mandatory/Optional for the Wallet|
|--------|-------------|---|---|
|secp256k1|ES256K|M|O*|
|Ed25519|EdDSA|M|O*|
|P-256 (secp256r1)|ES256|O|O|
|P-384 (secp384r1)|ES384|O|O|
* MUST support at least one.

**Review by Sakurann — CHANGES_REQUESTED** [review]

File: spec/spec.md (hunk 1 above)
Comment (Sakurann) [reply spec/spec.md:601]: also this sentence should be improved "While ION supports any public key JWK representation in a DID Document, implementors of this document MUST support JWT signature verification with the following Key Types and must support JWT signing with at least one." 
to something like
"Verifiers and wallets compliant with this profile MUST support JWT signature verification of the Key Types as defined in the table below"

**Comment by troyronda** [comment]
@Sakurann Updated.

File: spec/spec.md [hunk 2]
```diff
@@ -593,10 +593,14 @@ Below is a non-normative example of a DID Document that includes a serviceEndpoi
...
-|secp256k1|ES256K|
-|Ed25519|EdDSA|
+|Key Type|JWT Algorithm|Support to validate the signature by the Verifier and the Wallet|Support to sign by the Wallet|
+|--------|-------------|-------------|-------------|
+|secp256k1|ES256K|M|C*|
+|Ed25519|EdDSA|M|C*|
+|P-256 (secp256r1)|ES256|M|C*|
+|P-384 (secp384r1)|ES384|M|C*|
+
+`C*` Conditional: MUST support at least one.
```
Comment (jischr) [inline spec/spec.md:603]: What do you think about adding that M stands for must? 
```suggestion
`C*` Conditional: MUST support at least one.
`M` Must: MUST support
```

**Review by jischr — APPROVED** [review]

File: spec/spec.md (hunk 1 above)
Comment (Sakurann) [reply spec/spec.md]: this change has not been addressed

**Pull request merged**